### PR TITLE
Base generic view and queryset merging

### PR DESCRIPTION
**Pull request by tswicegood**
I've got a working `MergedNodesAndQuerySet` (open to new names) and the basic generic views for the wells.  Looking for some feedback.

### CODE REVIEW

**Comment by tswicegood** [comment]
Ignore the examples/admin commits that started this and are at the end.  It's in the `admin` branch and will come back.

File: armstrong/core/arm_wells/query.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        self.initialize()
+        total_in_well = len(self.well_content)
+        if j <= total_in_well:
+            return self.well_content[i:j]
+        end = j - total_in_well
+        if i >= total_in_well:
+            start = i - total_in_well
+            return self.queryset.exclude(pk__in=self.exclude_ids)[start:end]
+        return itertools.chain(self.well_content,
+                self.queryset.exclude(pk__in=self.exclude_ids)[0:end])
```
Comment (niran) [inline armstrong/core/arm_wells/query.py:47]: Consider a well with 10 nodes. `merged[5:15]` doesn't appear to work here at first glance. If `i < total_in_well` and `j > total_in_well`, it looks like all of `self.well_content` will be returned. There doesn't seem to be a test for that case.

Comment (tswicegood) [inline armstrong/core/arm_wells/query.py:47]: Good catch.  Adding another test case and the fix.

**Comment by tinio** [comment]
So far so good.  I am however trying to think of a name other than MergedNodesAndQuerySet

**Pull request merged**